bandersnatch/fp/sqrt.go

```diff
@@ -0,0 +1,293 @@
+package fp
+
+import "math/big"
+
+// The following code is _almost_ the original code from:
+// https://github.com/GottfriedHerold/Bandersnatch/blob/f665f90b64892b9c4c89cff3219e70456bb431e5/bandersnatch/fieldElements/field_element_square_root.go
+//
+// We had to do some changes to make it work with gnark:
+// - The type `feType_SquareRoot` was aliased to `Element` so everything looks the same. These types didn't have the exact
+//   same underlying representation, so it leaded to some minor adjustements. (e.g: accessing the limbs)
+// - Original APIs regarding finite-field multiplications (e.g: MulEq) were adjusted to use gnark Mul APIs.
+// - The original code had to explicitely do `Normalize()` after field element operations, but this isn't needed in gnark.
+// - The primitive 2^32-root-of unity value (see init()) was pulled from gnark FFT domain code.
+// - The original code used anonymous functions to define global vars, but we changed to use a init() function.
+//   This was required since we have other init() in the package that configure other globals (e.g: _modulus).
+//   By the way init() functions execution order works, we'll have these configured before the sqrt init() is called,
+//   compared with the original anonymous function global calls.
```

Comment:
I wrote a summary here of the work I did to adjust things to work for `go-ipa`. 
Sorry if sounds a bit cryptic. LMK any questions.